src/OpenFeature.DependencyInjection/MultiTarget/Guard.cs

```diff
@@ -0,0 +1,20 @@
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace OpenFeature;
+
+[DebuggerStepThrough]
+internal static class Guard
+{
+    public static T ThrowIfNull<T>(T? value, [CallerArgumentExpression("value")] string name = null!)
+    {
+#if NET8_0_OR_GREATER
+        ArgumentNullException.ThrowIfNull(value, name);
+#else
+        if (value is null)
+            throw new ArgumentNullException(name);
+#endif
+
```

Comment:
Personally I think it's better just do use the version in the #else here; using both a "nice" API and a "less nice" API is more complex than just using the "less nice" API, IMO.

Unless there some performance implication here I think this just hurts readability a bit.

Reply:
Fixed